reference_data/management/tests/update_dbnsfp_gene_tests.py

```diff
@@ -0,0 +1,67 @@
+    "MIR1302-2HG	.	.	.	.	.	107985730	.	XR_001737835	.	.	618355	MIR1302-2 host gene	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	1	1	1	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	0.0	0.0	0.0	0.0	0.0	0.0	0.0	0.0	.	.	.	.	.	.	.	.	.	.	.	.\n",
+    "A1BG-AS1	ENSG00000268895	19	NCRNA00181;A1BGAS;A1BG-AS	FLJ23569	.	.	503538	.	NR_015380	uc002qse.3	.	.	A1BG antisense RNA 1	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	1	1	1	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	.	7.987220447284345E-4	0.0	7.987220447284345E-4	0.0	0.02875399361022364	0.0	0.017172523961661343	0.0	.	.	.	.	.	.	.	.	.	.	.	.\n"
+]
+
+
+class UpdateDbNsfpGeneTest(TestCase):
+    fixtures = ['users', 'reference_data']
+    multi_db = True
+
+    def setUp(self):
+        # Create a temporary directory
+        self.test_dir = tempfile.mkdtemp()
+
+    def tearDown(self):
+        # Close the file, the directory will be removed after the test
+        shutil.rmtree(self.test_dir)
+
+    @mock.patch('reference_data.management.commands.utils.update_utils.logger')
+    @mock.patch('reference_data.management.commands.utils.download_utils.download_file')
+    def test_update_dbnsfp_gene_command(self, mock_download, mock_logger):
+        temp_file_path = os.path.join(self.test_dir, 'dbNSFP_gene')
+        with open(temp_file_path, 'w') as f:
+            f.write(u''.join(dbNSFP_GENE_DATA))
+        mock_download.return_value = temp_file_path
+
+        # test without a file_path parameter
+        call_command('update_dbnsfp_gene')
+
+        mock_download.assert_called_with('http://storage.googleapis.com/seqr-reference-data/dbnsfp/dbNSFP4.0_gene')
+
+        calls = [
+            mock.call('Deleting 3 existing dbNSFPGene records'),
+            mock.call('Parsing file'),
+            mock.call('Creating 1 dbNSFPGene records'),
+            mock.call('Done'),
+            mock.call('Loaded 1 dbNSFPGene records from {}. Skipped 1 records with unrecognized genes.'.format(temp_file_path)),
```

Comment:
To increase coverage, I removed the gene id of a record with a gene name "MIR1302-2HG". The current code will not count it as a record.
I updated the gene name to "MIR1302-2HG-W/O-GeneID". Hope it can be more clear.

Reply:
I wasn't asking about the skipped gene, I was asking about the fact that there are 3 rows in the data and  the logger here only mentions 2 of them, I want to know what happened to the third. You have `OR4F5` get created and `MIR1302-2HG-W/O-GeneID` get skipped (which was getting skipped before you changed the name and was clearer the way it was). What happened to `A1BG-AS1`?

Reply:
The gene `OR4F5` is loaded, `A1BG-AS1` is skipped, and the gene `MIR1302-2HG-W` is un-counted (neither loaded nor skipped) because it doesn't have a gene_id.

Reply:
Got it. It would be a bit more readable if you just took out the name for `MIR1302-2HG-W/O-GeneID` altogether and made it just a . and then changed A1BG-AS1 to something like `INVALID_GENE`

Reply:
`MIR1302-2HG-W/O-GeneID` and `A1BG-AS1` are both the gene_**_name_**. I would like a test without a gene_**_id_**. The gene `MIR1302-2HG-W/O-GeneID` has a gene_**_id_** `.` already. To make less confusing, I will change the gene_name `MIR1302-2HG-W/O-GeneID` to `INVALID-GENE-NO-ID` and `A1BG-AS1` to `INVALID_GENE-UNKNOWN`.